src/move/lp_coin_manager/sources/lp_coin_manager.move

```diff
@@ -0,0 +1,57 @@
+module lp_coin_manager::lp_coin_manager {
+    use aptos_framework::coin::{Self, Coin, BurnCapability, FreezeCapability, MintCapability};
+    use std::signer::{address_of};
+
+    #[resource_group = ObjectGroup]
+    struct LPCoinCapabilities<phantom CoinType, phantom LPCoinType> has key {
+        burn: BurnCapability<LPCoinType>,
+        mint: MintCapability<LPCoinType>,
+    }
+
+    // Intended to be called from `coin_factory`, this function facilitates
+    // storing the LP Coin capabilities atomically when the coin factory
+    // module is initialized.
+    public fun store_capabilities<CoinType, LPCoinType>(
```

Comment:
After finalizing and verifying the core algorithm flow, consider trying to merge caps into the market for less calls the the `borrow_global` func

Reply:
Just to be clear, we decided not to do this to avoid having to use generics in the `Market` resource, right? Or did you still want to do this

Reply:
Glad to follow up in a separate suggestion PR, I'll see what I can do now that we have most of the building blocks